src/Framework/ProjectStartedEventArgs.cs

```diff
@@ -391,7 +389,7 @@ internal override void WriteToStream(BinaryWriter writer)
                 foreach (var propertyPair in validProperties)
                 {
                     writer.Write((string)propertyPair.Key);
-                    writer.Write((string)propertyPair.Value);
+                    writer.Write((string?)propertyPair.Value ?? "");
```

Comment:
Is it better to write an empty string here rather than null? At first glance, I'd prefer writer.Write be able to take null strings.

Reply:
We have an extension method already, actually. WriteOptionalString writes 0 if it's null or 1 if not, then proceeds to write the string. If it previously threw an exception for null, though, I imagine there's no need to worry about it being null even semi-often, so this is probably better as-is.